src/components/Search/SearchRouter/SearchRouterContext.tsx

```diff
@@ -35,19 +39,29 @@ function SearchRouterContextProvider({children}: ChildrenProps) {
 
         // There are callbacks that live outside of React render-loop and interact with SearchRouter
         // So we need a function that is based on ref to correctly open/close it
-        const toggleSearchRouter = () => {
-            if (searchRouterDisplayedRef.current) {
+        // When on `/search` Page we instead focus the Input
+        const toggleSearch = () => {
+            const isUserOnSearchPage = isSearchTopmostCentralPane();
+
+            if (isUserOnSearchPage && searchPageFocusInputRef.current) {
+                searchPageFocusInputRef.current?.();
+            } else if (searchRouterDisplayedRef.current) {
                 closeSearchRouter();
             } else {
                 openSearchRouter();
             }
         };
 
+        const registerSearchPageInput = (focusSearchInput?: () => void) => {
```

Comment:
Let's add unregister method as well 😄 

```ts
      const unregisterSearchPageInput = () => {
            searchPageInputRef.current = undefined;
        };
```

Reply:
Also how about we store the input ref in the context instead of the whole callback, this way you cna call .focus or .blur on the input and the rest logic is in the component itself.